pandas/core/indexes/multi.py

```diff
@@ -2326,28 +2339,32 @@ def get_value(self, series, key):
             # We have to explicitly exclude generators, as these are hashable.
             raise InvalidIndexError(key)
 
-        def _try_mi(k):
-            # TODO: what if a level contains tuples??
-            loc = self.get_loc(k)
-
-            new_values = series._values[loc]
-            if is_scalar(loc):
-                return new_values
-
-            new_index = self[loc]
-            new_index = maybe_droplevels(new_index, k)
-            return series._constructor(
-                new_values, index=new_index, name=series.name
-            ).__finalize__(self)
-
         try:
-            return _try_mi(key)
+            loc = self.get_loc(key)
         except KeyError:
             if is_integer(key):
-                return series._values[key]
+                loc = key
             else:
                 raise
 
+        return self._get_values_for_loc(series, loc, key)
+
+    def _get_values_for_loc(self, series: "Series", loc, key):
+        """
+        Do a positional lookup on the given Series, returning either a scalar
+        or a Series.
```

Comment:
I would rename to _get_values_for_positional_loc to make this extra clear (I know for indexes we *only* have the concept of positional indexing, but still). follown ok.

also pls type / doc-string as much as possible

Reply:
> _get_values_for_positional_loc

good idea, will follow up